src/pyhf/infer/__init__.py

```diff
@@ -118,15 +126,31 @@ def hypotest(
     """
     init_pars = init_pars or pdf.config.suggested_init()
     par_bounds = par_bounds or pdf.config.suggested_bounds()
-    tensorlib, _ = get_backend()
-
-    calc = AsymptoticCalculator(data, pdf, init_pars, par_bounds, qtilde=qtilde)
+    # get fixed vals from the model
+    model_fixed_vals = [
+        (index, init)
+        for index, (init, is_fixed) in enumerate(
+            zip(init_pars, pdf.config.suggested_fixed())
+        )
+        if is_fixed
+    ]
+    # add user-defined ones at the end
+    fixed_vals = model_fixed_vals + (fixed_vals or [])
+
+    # de-dupe and use last-appended result for each index
+    fixed_vals = list(dict(fixed_vals).items())
+
+    calc = AsymptoticCalculator(
+        data, pdf, init_pars, par_bounds, fixed_vals, qtilde=qtilde
+    )
     teststat = calc.teststatistic(poi_test)
     sig_plus_bkg_distribution, b_only_distribution = calc.distributions(poi_test)
 
     CLsb = sig_plus_bkg_distribution.pvalue(teststat)
     CLb = b_only_distribution.pvalue(teststat)
     CLs = CLsb / CLb
+
+    tensorlib, _ = get_backend()
```

Comment:
Why is this setting of `tensorlib` needed? Sorry, just not immediately obvious.

Reply:
I moved it further down to where it's needed.